x-pack/plugins/lens/public/datatable_visualization/components/palette_panel_container.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import './palette_panel_container.scss';
+
+import React, { useState, useEffect, MutableRefObject } from 'react';
+import {
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiTitle,
+  EuiButtonIcon,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFocusTrap,
+  EuiOutsideClickDetector,
+  EuiPortal,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+export function PalettePanelContainer({
+  isOpen,
+  handleClose,
+  children,
+  siblingRef,
+}: {
+  isOpen: boolean;
+  handleClose: () => void;
+  children: React.ReactElement | React.ReactElement[];
+  siblingRef: MutableRefObject<HTMLDivElement | null>;
+}) {
+  const [focusTrapIsEnabled, setFocusTrapIsEnabled] = useState(false);
+
+  const closeFlyout = () => {
+    handleClose();
+    setFocusTrapIsEnabled(false);
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      // without setTimeout here the flyout pushes content when animating
+      setTimeout(() => {
+        setFocusTrapIsEnabled(true);
+      }, 255);
+    }
+  }, [isOpen]);
+
+  return isOpen && siblingRef.current ? (
+    <EuiPortal insert={{ sibling: siblingRef.current, position: 'after' }}>
```

Comment:
The clickable area for the palette is not very big, it seems like it's only 10px or so high. I was expecting the clickable area to be the whole area around the palette:

<img width="345" alt="Screen Shot 2021-05-14 at 11 11 23 AM" src="https://user-images.githubusercontent.com/666475/118297413-171bbb00-b4ac-11eb-98f5-ea7a6dd76a9e.png">

Reply:
I think that by design only the `Edit` button was the clickable way to open the panel.
I did expand it to be also the palette (although I find the regular cursor to be a bit confusing on the palette even if clickable). You mean the area should be even bigger than that? 🤔 

Maybe @MichaelMarcialis may give some feedback on this.

Reply:
I also found the regular cursor confusing. Overall, I was expecting that the palette was more like a "palette button" that would be a single clickable area. Is that possible?

Reply:
@wylieconlon, @dej611: I agree. Having both the "Edit" text/icon and the preceding `EuiColorPaletteDisplay` component be clickable and trigger the edit color interface would be nice usability change. I imagine hovering either should change the cursor to a `pointer`.